### DISCUSSION TITLE
Platform Suggestions

### DISCUSSION THREAD

**Discussion by WolfoTheWolf** [pelican-dev/panel#33] [Ideas]
Here are the suggestions I asked in the Stages channel. Ill post more suggestions as I think of them:

**Staff permissions:** 
- Being able to specify certain permissions for administrator users on the /admin side of the panel. Basically, one person cannot manage nodes, but another can.

**S3 Backup Management:** 
- Full front-end version of Wings backup system, able to do the following: 
  - Specify max server size: "Define the maximum size of the server in MB. If the server exceeds this size, it will be excluded from the backup".
  - Retention days: "Define the number of days to keep the backup".
  - Specify max backups being made at one time.
  - Specify nodes that are getting backed up.
  - Specify an S3 server and be able to create multiple S3 servers that can be assigned to nodes and can be run on a backup schedule individually using cron jobs, with S3 servers being able to manage URL lifespans, style endpoints, accelerated endpoints. There is a current addon for Ptero that allows this, but I would prefer this be a built-in feature as it's already utilized to some extent by Wings.

**Smart System Statistics:** 
- Ability to view node performance, storage usage, CPU usage, memory usage, and possibly live logs of Wings or general panel log files. These log files do not need to be live but maybe a button that we can push and auto download them. For server statistics this should be able to be filtered down by game server and by node. Maybe able to see logins, invalid logins and such just to be able to monitor everything in one spot.

**Multi Drive Support:** 
- This would allow using multiple drives. Instead of needing to run RAID or combined drives via software it would be really nice if we could specify specific drives for servers, example, NODE 1, can have access to drive 1 - 4 on that available system if configured, thus it could auto rollover to the other drives IF they are full, or we can manually set an active drive to be used so we don't have to have game server files spread across two drives. Would like this to be a front end configurable thing as well.

While yes, backups are available in Ptero but it would be nice if we could configure all of this on the front end of the panel, and some of the other configure items that are hidden in a config file within Ptero. Just to clean it all up a little more.

**Possible live system logs (Wings and Ptero)** 
- Which could be sent to Telegram, Discord, and emails. Now this would be both for the /admin page and the normal everyday server page. Users would be able to configure all server notifications to their preferred platform, server crashes and so on. But for administrators on the /admin page, we would be able to see if a Wings server encounters an error, or crashes, same with the general panel. If a game server crashes and for what reason, maybe invalid login attempts or login attempts as a whole. General stuff like that, while still maintaining full control over these options via the staff page. Would add yet another feature to the panel.

Overall love the idea of the project and these are just some things I would like to the panel. Some things may not be viable for all users, just throwing my two cents in to everything

Keep up the work and I am looking forward to see what comes of the project!

**Comment by purejosh** [comment]
You really should submit these as individual suggestions, and remove this one. I'm not voting for your entire list, and the reception of this massive list of suggestions is going to be abysmal, as people would need to open it up to see what you're actually suggesting.

**Reply by WolfoTheWolf** [reply]
Sounds good, didnt really think about individual ones. Said I did in the stages chat and I got a green check so I figured it was okay.

**Reply by purejosh** [reply]
All good, it'll just help us vote on what's most important to us.

**Reply by WolfoTheWolf** [reply]
Cant delete this so go ahead when you get a chance

**Comment by lancepioch** [comment]
Thanks for creating the individual discussions!
